spring-integration-gemfire/src/main/java/org/springframework/integration/gemfire/metadata/GemfireMetadataStore.java

```diff
@@ -82,4 +94,53 @@ public String remove(String key) {
 		return this.region.remove(key);
 	}
 
+	@Override
+	public void addListener(MetadataStoreListener listener) {
+		Assert.notNull(listener, "'listener' must not be null");
+		this.cacheListener.listeners.add(listener);
+	}
+
+	@Override
+	public void removeListener(MetadataStoreListener listener) {
+		this.cacheListener.listeners.remove(listener);
+	}
+
+	private static class GemfireCacheListener extends CacheListenerAdapter<String, String> {
+
+		private final List<MetadataStoreListener> listeners = new CopyOnWriteArrayList<MetadataStoreListener>();
+
+		GemfireCacheListener() {
+			super();
+		}
+
+		@Override
+		public void afterCreate(EntryEvent<String, String> event) {
+			for (MetadataStoreListener listener : this.listeners) {
+				listener.onAdd(event.getKey(), event.getNewValue());
+			}
+		}
+
+		@Override
+		public void afterUpdate(EntryEvent<String, String> event) {
+			for (MetadataStoreListener listener : this.listeners) {
+				listener.onUpdate(event.getKey(), event.getNewValue());
+			}
+		}
+
+		@Override
+		public void afterInvalidate(EntryEvent<String, String> event) {
```

Comment:
I think we shouldn't consider this operation  because `GemfireMetadataStore` doesn't support `invalidate`:
```
/**
   * Invalidates the entry with the specified key. Invalidate only removes the value from the entry,
   * the key is kept intact. To completely remove the entry, destroy should be used. The invalidate
   * will be distributed to other caches if the scope is not Scope.LOCAL.
...
   */
  public void invalidate(Object key) throws TimeoutException, **EntryNotFoundException;**
```